homeassistant/components/motion_blinds/__init__.py

```diff
@@ -220,7 +220,7 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         multicast.Unregister_motion_gateway(config_entry.data[CONF_HOST])
         hass.data[DOMAIN].pop(config_entry.entry_id)
 
-    if len(hass.data[DOMAIN]) == 3:
+    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
```

Comment:
Check the state too. See eg:
https://github.com/home-assistant/core/blob/2bb6e4bb877e21f2e49bcf66eddba18b66ff13ed/homeassistant/components/simplisafe/__init__.py#L428-L433

Reply:
Thanks, also implemented that